Makefile

```diff
@@ -30,35 +30,61 @@ endif
 
 all: manager
 
-# Run unit tests
+##@ Development
+
+fmt: ## Run go fmt against code
+	go fmt ./...
+
+vet: ## Run go vet against code
+	go vet ./...
+
+generate: controller-gen ## Generate code
+	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
+
 TEST ?= ./pkg/... ./api/... ./cmd/... ./controllers/... ./test/e2e/util/mongotester/...
-test: generate fmt vet manifests
+test: generate fmt vet manifests ## Run unit tests
 	go test $(TEST) -coverprofile cover.out
 
-# Build manager binary
-manager: generate fmt vet
+manager: generate fmt vet ## Build manager binary
 	go build -o bin/manager ./cmd/manager/main.go
 
-# Run against the configured Kubernetes cluster in ~/.kube/config
-run: install install-rbac
+run: install install-rbac ## Run against the configured Kubernetes cluster in ~/.kube/config
 	eval $$(scripts/dev/get_e2e_env_vars.py $(cleanup)); \
 	go run ./cmd/manager/main.go
 
 debug: install install-rbac
 	eval $$(scripts/dev/get_e2e_env_vars.py $(cleanup)); \
 	dlv debug ./cmd/manager/main.go
 
-# Install CRDs into a cluster
-install: manifests helm install-crd
+CONTROLLER_GEN = $(shell pwd)/bin/controller-gen
+controller-gen: ## Download controller-gen locally if necessary
+	$(call go-install-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1)
+
+# Try to use already installed helm from PATH
+ifeq (ok,$(shell test -f "$$(which helm)" && echo ok))
+    HELM=$(shell which helm)
+else
+    HELM=/usr/local/bin/helm
+endif
+
+helm: ## Download helm locally if necessary
+	$(call install-helm)
+
+install-prerequisites-macos: ## installs prerequisites for macos development
+	scripts/dev/install_prerequisites.sh
+
+##@ Installation/Uninstallation
+
+install: manifests helm install-crd ## Install CRDs into a cluster
 
 install-crd:
 	kubectl apply -f config/crd/bases/mongodbcommunity.mongodb.com_mongodbcommunity.yaml
 
 install-chart:
-	$(HELM) upgrade --install $(STRING_SET_VALUES) $(RELEASE_NAME_HELM) $(HELM_CHART)
+	$(HELM) upgrade --install $(STRING_SET_VALUES) $(RELEASE_NAME_HELM) $(HELM_CHART) --namespace $(NAMESPACE) --create-namespace
```

Comment:
While running it locally I had the case that I setup a namespace that did not exist in my `config.json`.
With this the operator was installed in the `default` namespace, while a lot of other commands still assumed the one from `config.json`. For instance `helm undeploy` fails with the namespace target